test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -4218,26 +4218,28 @@ Describe 'Invoke-WebRequest and Invoke-RestMethod support Cancellation through C
             [string]$Command = 'Invoke-WebRequest',
             [string]$Arguments = '',
             [uri]$Uri,
-            [int]$DelayMs = 100,
+            [int]$DelayBeforeStopSimulationMs = 5000,
             [switch]$WillComplete
         )
 
         $pwsh = [PowerShell]::Create()
         $invoke = "`$result = $Command -Uri `"$Uri`" $Arguments"
         $task = $pwsh.AddScript($invoke).InvokeAsync()
-        Start-Sleep -Milliseconds $DelayMs
+        $delay = [System.Threading.Tasks.Task]::Delay($DelayBeforeStopSimulationMs)
+
+        # Simulate CTRL-C as soon as the timeout expires or the main task ends
+        $null = [System.Threading.Tasks.Task]::WaitAny($task, $delay)
         $task.IsCompleted | Should -Be $WillComplete.ToBool()
         $pwsh.Stop()
         Wait-UntilTrue { [bool]($Task.IsCompleted) } | Should -BeTrue
```

Comment:
After `pwsh.Stop()` (simulating `ctrl+c`), this will wait for `$task` to be complete for upto 10 seconds (default timeout used), so even if this check passes, how can we be sure `$task` completes because of `ctrl+c` instead of "run till finish"?

Reply:
The stall duration is 30 seconds from the web server so the tasks would not complete unless the ctrl-c was successful.

Reply:
The `$uri = Get-WebListenerUrl -test Delay -TestValue 30` requests the web server to stall for 30 seconds after the start of the request has been written. It is coded to write out half of the total bytes, then delay by the stall amount before writing the rest.

Reply:
That's the missing part for me. Thanks!
Can you add a comment before the `Wait-UntilTrue` line about "the stall duration is 30 sections ..."? That will be helpful to someone who reads the code in future.

Reply:
Ok, added in last commit